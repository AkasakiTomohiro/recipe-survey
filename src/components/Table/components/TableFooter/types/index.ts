/**
 * @file index.ts
 * @author Artan's Projects
 * @copyright © 2024 Artan's Projects. All rights reserved.
 */

export interface ITableFooterProps {

  /**
   * 行数
   */
  numOfLines: number;

  /**
   * 1ページの行数
   */
  rowsPerPage: number;

  /**
   * ページ数
   */
  page: number;

  /**
   * ページネーション関数
   */
  onPageChange: (event: unknown, newPage: number) => void;

  /**
   * 1ページの行数を変更する関数
   */
  onRowsPerPageChange: (event: React.ChangeEvent<HTMLInputElement>) => void;

}

