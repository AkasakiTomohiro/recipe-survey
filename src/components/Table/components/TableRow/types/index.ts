/**
 * @file index.ts
 * @author Artan's Projects
 * @copyright © 2024 Artan's Projects. All rights reserved.
 */

export interface ITableRowProps {

  /**
   * 行番号
   */
  index: number;

  /**
   * 1行のデータ
   */
  row: (number | string)[];
}
