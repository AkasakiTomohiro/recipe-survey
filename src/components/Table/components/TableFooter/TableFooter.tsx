/**
 * @file TableFooter.tsx
 * @author Artan's Projects
 * @copyright Â© 2024 Artan's Projects. All rights reserved.
*/
import { TablePagination } from '@mui/material';

import { ITableFooterProps } from './types';

export const TableFooter = (props: ITableFooterProps): JSX.Element => {
  return (
    <TablePagination
      rowsPerPageOptions={[10, 25, 100]}
      component="div"
      count={props.numOfLines}
      rowsPerPage={props.rowsPerPage}
      page={props.page}
      onPageChange={props.onPageChange}
      onRowsPerPageChange={props.onRowsPerPageChange}
    />
  );
};
