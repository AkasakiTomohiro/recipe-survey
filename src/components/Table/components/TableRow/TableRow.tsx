/**
 * @file TableRow.tsx
 * @author Artan's Projects
 * @copyright Â© 2024 Artan's Projects. All rights reserved.
 */

import TableCell from '@mui/material/TableCell';
import MuiTableRow from '@mui/material/TableRow';

import { ITableRowProps } from './types';

export const TableRow = ({ row, index }: ITableRowProps): JSX.Element => {
  return (
    <MuiTableRow hover>
      {row.map((m, i) => (<TableCell key={`table-row-${index}-cell-${i}`}>{m}</TableCell>))}
    </MuiTableRow>
  );
};
