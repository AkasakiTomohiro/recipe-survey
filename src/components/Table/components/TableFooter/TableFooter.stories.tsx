/**
 * @file TableFooter.stories.tsx
 * @author Artan's Projects
 * @copyright Â© 2024 Artan's Projects. All rights reserved.
 */

import type { Meta, StoryObj } from '@storybook/react';

import { Table, TableBody } from '@mui/material';
import Paper from '@mui/material/Paper';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableRow from '@mui/material/TableRow';

import { TableFooter } from './TableFooter';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title     : 'components/Table/TableFooter',
  component : TableFooter,
  parameters: {

    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered'
  },

  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs']
} satisfies Meta<typeof TableFooter>;
export default meta;

type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
export const Primary: Story = {
  args: {
    numOfLines: 12,
    rowsPerPage: 10,
    page: 0,
    onPageChange: () => console.log("change page"),
    onRowsPerPageChange: () => console.log("change rows per page"),
  },
  decorators: [Story => (
    <Paper>
      <TableContainer>
        <Table>
          <TableBody>
            {[...Array(10)].map((m, i) => (
            <TableRow key={i}>
              <TableCell>Value1</TableCell>
              <TableCell>Value2</TableCell>
              <TableCell>Value3</TableCell>
              <TableCell>Value4</TableCell>
              <TableCell>Value5</TableCell>
              <TableCell>Value6</TableCell>
            </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
      <Story />
    </Paper>
  )]
};
