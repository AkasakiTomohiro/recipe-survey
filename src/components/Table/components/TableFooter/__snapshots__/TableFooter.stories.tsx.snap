// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Table/TableFooter Primary smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-okvxkk-MuiPaper-root">
  <div class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root">
    <table class="MuiTable-root css-qzpwgq-MuiTable-root">
      <tbody class="MuiTableBody-root css-apqrd9-MuiTableBody-root">
        <tr class="MuiTableRow-root css-dvunen-MuiTableRow-root">
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value1
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value2
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value3
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value4
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value5
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value6
          </td>
        </tr>
        <tr class="MuiTableRow-root css-dvunen-MuiTableRow-root">
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value1
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value2
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value3
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value4
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value5
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value6
          </td>
        </tr>
        <tr class="MuiTableRow-root css-dvunen-MuiTableRow-root">
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value1
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value2
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value3
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value4
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value5
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value6
          </td>
        </tr>
        <tr class="MuiTableRow-root css-dvunen-MuiTableRow-root">
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value1
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value2
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value3
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value4
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value5
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value6
          </td>
        </tr>
        <tr class="MuiTableRow-root css-dvunen-MuiTableRow-root">
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value1
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value2
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value3
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value4
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value5
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value6
          </td>
        </tr>
        <tr class="MuiTableRow-root css-dvunen-MuiTableRow-root">
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value1
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value2
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value3
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value4
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value5
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value6
          </td>
        </tr>
        <tr class="MuiTableRow-root css-dvunen-MuiTableRow-root">
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value1
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value2
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value3
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value4
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value5
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value6
          </td>
        </tr>
        <tr class="MuiTableRow-root css-dvunen-MuiTableRow-root">
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value1
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value2
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value3
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value4
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value5
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value6
          </td>
        </tr>
        <tr class="MuiTableRow-root css-dvunen-MuiTableRow-root">
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value1
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value2
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value3
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value4
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value5
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value6
          </td>
        </tr>
        <tr class="MuiTableRow-root css-dvunen-MuiTableRow-root">
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value1
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value2
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value3
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value4
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value5
          </td>
          <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-16blcct-MuiTableCell-root">
            Value6
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div class="MuiTablePagination-root css-uxtgd5-MuiTablePagination-root">
    <div class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar">
      <div class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer">
      </div>
      <p class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
         id=":r4:"
      >
        Rows per page:
      </p>
      <div class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-input css-9fvjej-MuiInputBase-root-MuiTablePagination-select">
        <div tabindex="0"
             role="combobox"
             aria-controls=":r5:"
             aria-expanded="false"
             aria-haspopup="listbox"
             aria-labelledby=":r4: :r3:"
             id=":r3:"
             class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
        >
          10
        </div>
        <input aria-invalid="false"
               aria-hidden="true"
               tabindex="-1"
               class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
               value="10"
        >
        <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
             focusable="false"
             aria-hidden="true"
             viewbox="0 0 24 24"
             data-testid="ArrowDropDownIcon"
        >
          <path d="M7 10l5 5 5-5z">
          </path>
        </svg>
      </div>
      <p class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows">
        1–10 of 12
      </p>
      <div class="MuiTablePagination-actions">
        <button class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                tabindex="-1"
                type="button"
                disabled
                aria-label="Go to previous page"
                title="Go to previous page"
        >
          <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
               focusable="false"
               aria-hidden="true"
               viewbox="0 0 24 24"
               data-testid="KeyboardArrowLeftIcon"
          >
            <path d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z">
            </path>
          </svg>
        </button>
        <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
                aria-label="Go to next page"
                title="Go to next page"
        >
          <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
               focusable="false"
               aria-hidden="true"
               viewbox="0 0 24 24"
               data-testid="KeyboardArrowRightIcon"
          >
            <path d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z">
            </path>
          </svg>
          <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
